apps/api-journeys/src/modules/block-response/index.ts

```diff
@@ -0,0 +1,69 @@
+import 'reflect-metadata'
+import { createModule, gql } from 'graphql-modules'
+import { BlockResponseModule } from './__generated__/types'
+
+const typeDefs = gql`
+  enum VideoBlockResponseStateEnum {
+    PLAYING
+    PAUSED
+    FINISHED
+  }
+
+  extend type Mutation {
+    journeySessionCreate(journeyId: ID!): ID!
+
+    signupBlockResponseCreate(journeySessionId: ID!, blockId: ID!, name: String!, email: String!): ID!
+    videoBlockResponseCreate(journeySessionId: ID!, blockId: ID!, position: Float!, state: VideoBlockResponseStateEnum!): ID!
+    radioQuestionBlockResponseCreate(journeySessionId: ID!, blockId: ID!, selectedResponseBlockId: ID!): ID!
+  }
+`
+
+const resolvers: BlockResponseModule.Resolvers = {
+  Mutation: {
+    async journeySessionCreate (_parent, { journeyId }, { db }) {
+      const session = await db.journeySession.create({
+        data: {
+          journeyId
+        }
+      })
+      return session.id
+    },
+    async signupBlockResponseCreate (_parent, { journeySessionId, blockId, name, email }, { db }) {
+      const response = await db.blockResponse.create({
+        data: {
+          journeySessionId,
+          blockId,
+          responseData: { name, email }
+        }
+      })
+      return response.id
```

Comment:
I like to leverage destructuring to be explicit about what is being used in the response. Not needed though if you don't like it.

```suggestion
      const { id } = await db.blockResponse.create({
        data: {
          journeySessionId,
          blockId,
          responseData: { name, email }
        }
      })
      return id
```

Reply:
Awesome, thanks for the reminder on this, I forget about destructuring in TS